### PR TITLE
Non hardcoded popups & toggle addons through popup (responsive iframe)

### LINKED ISSUE #521: Toggle addons through the extension popup
Would be nice

### LINKED ISSUE #773: Addon popups should not be hardcoded
(Moved from #15 and #16) 

Currently (since v1.0.0 until now) popups are "hardcoded". This means that:
1) Addons can't specify details about their popup on their manifest
2) Users can't remove/reorder/etc. popups
3) Addons have no standard way to send messages from/to popups
4) Addons don't actually have `addon.*` APIs, and localization works weirdly on them

### LINKED ISSUE #1054: Cloud games popups: links to users don't work
They're \<a>'s without href's :P

### PR DESCRIPTION
**Pull request by TheColaber**
**Resolves**

Resolves #773
 - completes number one
 - completes number two
 - 3 and 4 yet to be done if even necessary
 
Resolves #1054
Soon to resolve #521

**Changes**

added a new addon.json property: "popup"
removed cloud-games notice
change popups system
added vue to popup
removed the now unnecessary error code for scratch-messaging
added links to users in cloud-games
made settings even more responsive for iframes
made a new popup for quick settings

**Tests**
:O WE ARE READY TO MERGE :O

Definitely, definitely, definitely, we will need a lot of testing on this PR after this last todo is done. I have a feeling this PR will be open for a while...

### CODE REVIEW

**Comment by TheColaber** [comment]
cc @Hans5958 @apple502j

File: addons/cloud-games/addon.json [hunk 1]
```diff
@@ -10,6 +9,11 @@
       "name": "griffpatch"
     }
   ],
+  "popup": {
+    "addonId": "cloud-games",
```
Comment (apple502j) [inline addons/cloud-games/addon.json:13]: Seems unnecessary.

Comment (TheColaber) [reply addons/cloud-games/addon.json:13]: Including the id? I guess so... I guess I'll code a bit more to remove it.

**Comment by GrahamSH-LLK** [comment]
@TheColaber It would be nice if we could use vue on the popup, for ease of development. You could take a look at the settings page to figure that out.

**Review by D1WK — APPROVED** [review]

**Comment by Explosion-Scratch** [comment]
Maybe instead of just empty stuff, you can put a placeholder like ```Turn on the Scratch Messaging extension to view your messages here!```

**Comment by WorldLanguages** [comment]
TODO:
1. Localize popup name
2. Dynamic "iframe" class in settings page
3. Settings page is currently hardcoded in order to test #521 
4. Remove overflow-x:hidden when fixing 5
5. Make addon settings look good and not overflow. Test all setting types
6. Add settings pseudoaddon
7. Remove "no enabled popup addons" messages, styles, etc. since this pseudoaddon wouldn't be able to get disabled
8. Test settings through popup with different zoom levels
9. If popup clicked on a Scratch tab, show first addons that are running on that tab

File: webpages/settings/index.html [hunk 2]
```diff
@@ -382,5 +382,11 @@ <h1 class="setting-heading">{{ msg("moreSettings") }}</h1>
         </div>
       </div>
     </div>
+    <!-- <div class="popup">
```
Comment (WorldLanguages) [inline webpages/settings/index.html:385]: TODO: remove?

Comment (WorldLanguages) [reply webpages/settings/index.html:385]: TODO: remove CSS as well

File: content-scripts/cs.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ function onHeadAvailable({ globalState, l10njson, addonsWithUserscripts, usersty
 
   chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     if (request === "getRunningAddons") {
-      sendResponse({...addonsWithUserscripts, ...themes })
+      sendResponse({ ...addonsWithUserscripts, ...themes });
```
Comment (WorldLanguages) [inline content-scripts/cs.js:126]: TODO: make sure it works properly

File: webpages/settings/index.html (hunk 2 above)
Comment (WorldLanguages) [reply webpages/settings/index.html]: Not removed, used this modal to fix todo 13

**Pull request merged**